src/services/dashboard-service.js

```diff
@@ -75,27 +84,42 @@ class DashboardService {
           return {
             type: USER_ACTIVITY_TYPE.documentMarkedFavorite,
             timestamp: favorite.setOn,
-            getData: async () => {
+            data: null,
+            isDeprecated: null,
+            [completionFunction]: async () => {
               const document = await this.documentStore.getDocumentMetadataByKey(favorite.id);
-              return { _id: favorite.id, title: document.title };
+              return {
+                data: { _id: favorite.id, title: document.title },
```

Comment:
We get a `cannot read properties of undefined` exception here if `document` is `null`, and this can occur if the favorite document has source `external` because external documents can be hard deleted by admins.